src/index.js

```diff
@@ -0,0 +1,71 @@
+/** Copyright (c) 2017 Uber Technologies, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const requiredLabels = [
+  'breaking',
+  'feature',
+  'bugfix',
+  'docs',
+  'discussion',
+  'release',
+  'prerelease',
+  'greenkeeping',
+];
+
+module.exports = robot => {
+  robot.on('pull_request.opened', check);
+  robot.on('pull_request.reopened', check);
+  robot.on('pull_request.edited', check);
+  robot.on('pull_request.synchronize', check);
+  robot.on('pull_request.unlabeled', check);
+  robot.on('pull_request.labeled', check);
+
+  async function check(context) {
+    const pr = context.payload.pull_request;
+
+    // set status to pending while checks happen
+    setStatus(context, {
+      state: 'pending',
+      description:
+        'Checking whether at least one required semver-related label exists',
+    });
+
+    function getRequiredLabels(pr) {
+      const labels = pr.labels;
+      const filteredLabels = labels.filter(l =>
+        requiredLabels.includes(l.name),
+      );
```

Comment:
For v2, it might be interesting to make this even more generic and support arbitrary rules, such as multiple label enums, etc.

I think a good pattern for this sort of stuff might be:
- Several probots that analyze code and apply/remove labels (sort of like `probot-app-label-release-pr` based on certain heuristics. You could think of this as "tagging" bots.
- One probot (this probot) which applies generic label rules. This is the "enforcer" bot.

Perhaps it could take some JSON/YAML rule DSL which would define label requirements.

Super poor strawman proposal:
```
[
  {
    "require": {
      "oneOf": ["semver-minor", "semver-major", "semver-patch"]
    },
    "except": {
      "oneOf": ["docs", "infrastructure"]
    }
  }
]
```

It wouldn't surprise me if there's some JSON rule DSL we could use out of the box.

Or if a DSL is a pain, maybe just a JS library for building these label enforcer bots.